styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java

```diff
@@ -114,11 +117,22 @@ private void tryTriggering(Workflow workflow,
               instantSpec.instant());
           // Wait for the event to be processed before proceeding to the next trigger
           processed.toCompletableFuture().get();
-        } catch (AlreadyInitializedException ignored) {
-          // nop
-        } catch (Throwable e) {
-          LOG.warn("Triggering {} threw exception", workflow.id(), e);
-          return; // so we don't update the trigger time
+        } catch (ExecutionException e) {
```

Comment:
Are we sure this will never be further wrapped in another exception? I think to more properly mitigate fragility we could just `catch (Exception e)`?

Reply:
It shouldn't or we will need to do the `findCause` stuff.

Reply:
Yeah I mean we just do the `findCause` stuff for `Exception`, instead of having two branches, one for `Exception` that doesn't do `findCause` stuff, and one for `ExecutionException` that does `findCause` stuff and duplicates `Exception` stuff.

Reply:
"It shouldn't" implies relying on humans that they won't wrap it further, that's what I'm worried about.

Reply:
Exactly so we are on it.